src/openrct2/ride/ShopItem.h

```diff
@@ -98,27 +93,39 @@ struct ShopItemDescriptor
     uint32_t Image;
     ShopItemStrings Naming;
     uint16_t Flags;
+    uint8_t LitterType;
+    uint8_t ConsumptionTime;
+    ShopItem DiscardContianer;
+    PeepThoughtType TooMuchThought;
+    PeepThoughtType GoodValueThought;
 
-    bool HasFlag(const uint16_t flag) const;
+    constexpr bool HasFlag(const uint16_t flag) const
```

Comment:
Why moving this to the header?

Reply:
constexpr need them in the header. Its the reason why i've got a bit of a convoluted thing to get all the flags out of the array.